pandas/io/sas/sasreader.py

```diff
@@ -29,8 +29,11 @@ def read_sas(filepath_or_buffer, format=None, index=None, encoding=None,
     DataFrame if iterator=False and chunksize=None, else SAS7BDATReader
     or XportReader
     """
-
+    from pandas import compat
     if format is None:
+        buffErr = "Format unrecognized. If buffer object, specify format"
```

Comment:
use buffer_error_msg.

If this is a buffer object, rather than a string name, you must specify a format string.